### PR TITLE
Add ConstraintVerifier section to Spring Boot Quickstart guide

### PR DESCRIPTION
**Pull request by Christopher-Chianelli**
Basically a copy of https://github.com/quarkusio/quarkus/pull/15611 ; any changes to this should also apply to that.

### Checklist
- [ ] Documentation updated if applicable.
- [ ] Upgrade recipe provided if applicable.

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* for a <b>pull request build</b> please add comment: <b>Jenkins retest this</b>
* for a <b>full downstream build</b> please add comment: <b>Jenkins run fdb</b>
* for a <b>compile downstream build</b> please add comment: <b>Jenkins run cdb</b>
* for a <b>full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>
* for an <b>upstream build</b> please add comment: <b>Jenkins run upstream</b>
* for a <b>Quarkus LTS check</b> please add comment: <b>Jenkins run LTS</b>
* for a <b>Native check</b> please add comment: <b>Jenkins run native</b>
</details>

### CODE REVIEW

File: optaplanner-docs/src/main/asciidoc/QuickStart/SpringBoot/SpringBootQuickStart-chapter.adoc [hunk 1]
```diff
@@ -881,6 +881,64 @@ This avoids hard coding a solver time, because the unit test might run on arbitr
 This approach ensures that the test runs long enough to find a feasible solution, even on slow machines.
 But it does not run a millisecond longer than it strictly must, even on fast machines.
 
+Additionally, if you use a `ConstraintProvider` and have `org.optaplanner:optaplanner-test` as a dependency in your pom, you can test per constraint using a
+`ConstraintVerifier`.
+
+Add the `org.optaplanner:optaplanner-test` dependency to your `pom.xml` dependencies section:
```
Comment (triceo) [inline optaplanner-docs/src/main/asciidoc/QuickStart/SpringBoot/SpringBootQuickStart-chapter.adoc:887]: This doesn't make much sense. First you say you need to have `optaplanner-test` in your POM, then you tell people to add it there.

Also, consider that we also support Gradle. Personally, I wouldn't bother with the POM example - just mention the `org.optaplanner:optaplanner-test` GAV.

File: optaplanner-docs/src/main/asciidoc/QuickStart/SpringBoot/SpringBootQuickStart-chapter.adoc [hunk 2]
```diff
@@ -881,6 +881,64 @@ This avoids hard coding a solver time, because the unit test might run on arbitr
...
+[source,xml]
+----
+    <dependency>
+      <groupId>org.optaplanner</groupId>
+      <artifactId>optaplanner-test</artifactId>
+      <scope>test</scope>
+    </dependency>
+----
+
+Create the `src/test/java/com/example/solver/TimeTableConstraintProviderTest.java` class:
```
Comment (triceo) [inline optaplanner-docs/src/main/asciidoc/QuickStart/SpringBoot/SpringBootQuickStart-chapter.adoc:897]: Personally, I would reduce it to just the fields and the test method - the boilerplate does more bad than good, IMO. I think it is safe to assume that people know how to write a test.

**Review by triceo — CHANGES_REQUESTED** [review]
I think we can stil improve this.

File: optaplanner-docs/src/main/asciidoc/QuickStart/SpringBoot/SpringBootQuickStart-chapter.adoc (hunk 2 above)
Comment (Christopher-Chianelli) [reply optaplanner-docs/src/main/asciidoc/QuickStart/SpringBoot/SpringBootQuickStart-chapter.adoc:897]: That go against the style of the rest of the guide; the guide shows the full code, even getters and setters (i.e. can be copied and pasted into an IDE and work).

Comment (triceo) [reply optaplanner-docs/src/main/asciidoc/QuickStart/SpringBoot/SpringBootQuickStart-chapter.adoc:897]: OK, I grudgingly accept.

File: optaplanner-docs/src/main/asciidoc/QuickStart/SpringBoot/SpringBootQuickStart-chapter.adoc (hunk 1 above)
Comment (triceo) [reply optaplanner-docs/src/main/asciidoc/QuickStart/SpringBoot/SpringBootQuickStart-chapter.adoc:887]: I do think it belongs here, as this is the section about testing. I think we can avoid the entire Maven/Gradle question by just mentioning the GAV.

**Review by triceo — APPROVED** [review]

**Pull request merged**